### PR TITLE
GH-44797: [CI] Update pkg-config to pkgconf on Homebrew

### PR DESCRIPTION
**Pull request by raulcd**
### Rationale for this change

Some CI jobs are currently failing on main:

- [Python / ARM64 macOS 14 Python 3](https://github.com/apache/arrow/actions/runs/11936033926/job/33268789966?pr=44796)
- [C++ / ARM64 macOS 14 C++ ](https://github.com/apache/arrow/actions/runs/11935481942/job/33268873278)

### What changes are included in this PR?

Update from deprecated `pkg-config` to `pkgconf`

### Are these changes tested?

Via CI

### Are there any user-facing changes?

No
* GitHub Issue: #44797

### CODE REVIEW

**Comment by raulcd** [comment]
@kou any idea?

**Comment by assignUser** [comment]
@raulcd ruby uses `pkg-conf` via a Gem by kou and it looks like it can't find the new pkgconf. I think you can fix it by setting the envvar https://github.com/ruby-gnome/pkg-config/blob/main/lib/pkg-config.rb#L75

**Comment by assignUser** [comment]
For python I think it's because `pkg-config` is already installed via brew on the runner so it fails linking, removing `pkg-config` before should fix it: https://github.com/apache/arrow/actions/runs/11936401942/job/33269969915?pr=44798#step:4:65

**Comment by kou** [comment]
I'll improve the pkg-config gem.

File: .github/workflows/ruby.yml [hunk 1]
```diff
@@ -159,6 +159,7 @@ jobs:
           brew bundle --file=c_glib/Brewfile
       - name: Install Ruby Dependencies
         run: |
+          export PKG_CONFIG=/opt/homebrew/bin/pkg-config
```
Comment (kou) [inline .github/workflows/ruby.yml:162]: I've updated pkg-config gem. Let's try it.

```suggestion
```

**Comment by kou** [comment]
Hmm. Meson still uses only `pkg-config`: https://github.com/mesonbuild/meson/issues/7701

**Review by raulcd — COMMENTED** [review]
:+1:  Thanks @kou for the fixes!

**Review by kou — APPROVED** [review]
+1

**Pull request merged**